### PR TITLE
chore: Don't use "upsert" term in error messages

### PR DESCRIPTION
**Pull request by warber**
#### What this PR does / Why we need it:
#982
#### Special notes for your reviewer:
PR contains also a bit of cleanup/housekeeping.
Also, I've decided to get rid of "SendWithRetryWithInitialTry" and made it the default behavior.

#### Does this PR introduce a user-facing change?

Error logs/messages do not contain the term "upsert"

### CODE REVIEW

File: pkg/client/dtclient/client.go [hunk 1]
```diff
@@ -402,18 +402,18 @@ func (d *DynatraceClient) upsertSettings(obj SettingsObject) (DynatraceEntity, e
...
-		return DynatraceEntity{}, fmt.Errorf("failed to build settings object for upsert: %w", err)
+		return DynatraceEntity{}, fmt.Errorf("failed to build settings object: %w", err)
 	}
 
 	requestUrl := d.environmentURL + d.settingsObjectAPIPath
 
 	resp, err := rest.SendWithRetryWithInitialTry(d.client, rest.Post, obj.Id, requestUrl, payload, d.retrySettings.Normal)
 	if err != nil {
-		return DynatraceEntity{}, fmt.Errorf("failed to upsert dynatrace obj: %w", err)
+		return DynatraceEntity{}, fmt.Errorf("failed to create dynatrace obj: %w", err)
```
Comment (UnseenWizzard) [inline pkg/client/dtclient/client.go:412]: This is a "create or update" though, unlike the Config APIs where we can differentiate between create and update and log them differently. 

"Upsert" actually reads better to me than "create/update", but I think that would be thing to log after the change

File: pkg/rest/retry.go [hunk 2]
```diff
@@ -77,6 +77,10 @@ func GetWithRetry(client *http.Client, url string, settings RetrySetting) (resp
 
 // SendWithRetry will retry to call sendWithBody for a given number of times, waiting a give duration between calls
 func SendWithRetry(client *http.Client, sendWithBody SendRequestWithBody, objectName string, path string, body []byte, setting RetrySetting) (resp Response, err error) {
+	resp, err = sendWithBody(client, path, body)
```
Comment (UnseenWizzard) [inline pkg/rest/retry.go:80]: IIRC the Config Client calls to this happen after a first try failed (and a special retry case occurred)
While the Settings client goes straight to send with retry. 

Just changing this without modifying the handling for Config client errors, means we do an extra send/retry in their case right? 
I think that's visible from your change to the request_test assert as well, which expects +1 call. 

I don't think we should just do that, it kinda breaks the max retries by now always doing +1 "retry" call for Configs as they had an original API call before.

File: pkg/rest/retry.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ func GetWithRetry(client *http.Client, url string, settings RetrySetting) (resp
 	}
 
 	for i := 0; i < settings.MaxRetries; i++ {
-		log.Warn("Retrying failed GET request %s with error (HTTP %d)", url, resp.StatusCode)
+		log.Warn("Failed to get config %s with error (HTTP %d). Waiting for %s before retrying...", url, resp.StatusCode, settings.WaitTime)
```
Comment (UnseenWizzard) [inline pkg/rest/retry.go:63]: nit: These are not logging the `config` but the `url` - I think logging 

> Failed to get config https://my-end.live.dynatrace.com/api/something/something/uuid ..

reads a bit strange. 

Note that the Send one you're attempting to unify with, actually logs the `objectName`- maybe that's possible here as well. Maybe both the name and API call could be helpful also

**Review by UnseenWizzard — CHANGES_REQUESTED** [review]
I think some of the logging changes make the log more misleading than it was before, please take a look at the comments

**Comment by warber** [comment]
@UnseenWizzard I've dropped the questionable commits, pls rereview

File: pkg/client/dtclient/client.go [hunk 4]
```diff
@@ -402,18 +402,18 @@ func (d *DynatraceClient) upsertSettings(obj SettingsObject) (DynatraceEntity, e
...
 
 	resp, err := rest.SendWithRetryWithInitialTry(d.client, rest.Post, obj.Id, requestUrl, payload, d.retrySettings.Normal)
 	if err != nil {
-		return DynatraceEntity{}, fmt.Errorf("failed to upsert dynatrace obj: %w", err)
+		return DynatraceEntity{}, fmt.Errorf("failed to create or update dynatrace obj: %w", err)
 	}
 
 	if !success(resp) {
-		return DynatraceEntity{}, fmt.Errorf("failed to upsert settings object with externalId %s (HTTP %d)!\n\tResponse was: %s", externalId, resp.StatusCode, string(resp.Body))
+		return DynatraceEntity{}, fmt.Errorf("failed to create settings object with externalId %s (HTTP %d)!\n\tResponse was: %s", externalId, resp.StatusCode, string(resp.Body))
```
Comment (UnseenWizzard) [inline pkg/client/dtclient/client.go:416]:
```suggestion
		return DynatraceEntity{}, fmt.Errorf("failed to create or update settings object with externalId %s (HTTP %d)!\n\tResponse was: %s", externalId, resp.StatusCode, string(resp.Body))
```

Shouldn't this also be create or update @warber ?

**Review by UnseenWizzard — APPROVED** [review]
Thanks for addressing the multiple rounds of comments! 🚀

**Pull request merged**